docs/manual/scala/guide/cluster/DomainModelling.md

```diff
@@ -194,29 +194,93 @@ case class OpenShoppingCart(items: Map[String, Int]) extends ShoppingCart {
+
 ### Tagging the events - Akka Persistence Query considerations
 
-// TODO explain tagging
+Events are persisted in the event journal and are primarily used to replay the state of the aggregate each time it needs to be instantiated. However, in CQRS, we also want to consume those same events and generate read-side views or publish them in a message broker (eg: Kafka) for external consumption.
+
+To be able to consume the events on the read-side, the events must be tagged. Moreover, Lagom supports distributed journal consumption for Read-Side Processors and Topic Producers (aka: Projections) if you chose to use sharded tags.
+
+First we need to define how the events will be tagged. This is done using the `AggregateEventTag` utility. It's recommended to shard the tags so they can be consumed in a distributed fashion. You can also decided to not shard them as explained [here](https://www.lagomframework.com/documentation/current/scala/ReadSide.html#Event-tags).
+
+This example uses a shard of 10 and defines the event tagger in the companion object of `ShoppingCartEvent`. Note that tags must be stable as well as the number of shards. This can't be changed later without migrating the journal.
+
 ```scala
 object ShoppingCartEvent {
   val Tag = AggregateEventTag.sharded[ShoppingCartEvent](numShards = 10)
 }
 ```
 
+The `AggregateEventTag` is a Lagom class used by Lagom's [Read-Side Processor](https://www.lagomframework.com/documentation/current/scala/ReadSide.html) and [Topic Producers](https://www.lagomframework.com/documentation/current/scala/MessageBrokerApi.html#Implementing-a-topic), however Akka Persistence Typed expects a function `Event => Set[String]`. Therefore, we need to use an adapter to transform Lagom's `AggregateEventTag` to the required Akka tagger function.
+
+```scala
+EventSourcedBehavior
+  .withEnforcedReplies[ShoppingCartCommand, ShoppingCartEvent, ShoppingCart](
+    persistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId),
+    emptyState = OpenShoppingCart(Map.empty),
+    commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+    eventHandler = (cart, evt) => cart.applyEvent(evt)
+  ).withTagger(AkkaTaggerAdapter.fromLagom(entityContext, ShoppingCartEvent.Tag))
+```
+
+> Here again, we need the `entityContext`. This will be introduced on the next section.
+
 ### Configuring snaptshots
 
+Snapshotting is a common optimization to avoid replaying all the events since the beginning.
+
+You can define snapshots rules in two ways: by predicate and by counter. Both can be combined. The example bellow uses both methods to illustrate the APIs. Once again, you can find more details on the [Akka documentation](https://doc.akka.io/docs/akka/2.6/typed/persistence-snapshot.html).
```

Comment:
```suggestion
You can define snapshots rules in two ways: by predicate and by counter. Both can be combined. The example below uses both methods to illustrate the APIs. Once again, you can find more details on the [Akka documentation](https://doc.akka.io/docs/akka/2.6/typed/persistence-snapshot.html).
```

Reply:
Why demo two different ways of snapshotting?

We're already linking to the Akka docs. I think it'd be better to only demo the `withRetention` strategy which is the most similar to Lagom's classic Persistence approach to snapshotting (thought I'm not sure it's identical).